jack-and-the-beans/buddies-ios
# Bud 129 discover query

Adds
* Discover page! 
    * This sorts activities in [Algolia's default ranking order](https://www.algolia.com/doc/guides/managing-results/must-do/custom-ranking/) _but_
    * We group all activities by distance from the user.  Right now, if the user sets search range to 200miles, he will see things sorted by how many of his favorited topics they match for the first 66 miles.  Then he sees everything 66-133miles, sorted by # matching topics.  etc.
      -  This is the best we can do with relative easy using Algolia, since they rank with tie-breaking.  If we sort by granular distance, none of our other ranking metrics wil apply
      - If we sort by most matched topics and secondarily search on location, I think we lose the emphasis on locality in our app.
* Removed the Algolia search logic from the FilterComponent and put it into the ActivityVC
    * It makes sense for the VCs to control it, but as a result there is a bit of very similar code in both the `DiscoverTableVC` and `TopicActivityTableV` in `fetchAndLoadActivities`.  


How to test (Really hard to test, but here's a few things you can try):
* Try a couple different radius/time range filters on Discover
    * Check out the activities
    * Does it look like further activities are further down?
    * Are locations in the same proximity ranked by # of matched topics?*
* You can put `query.getRankingInfo = true` near line `116` of `AlgoliaSearch.swift` and then inspect the logs' response for your query on the Algolia dashboard and make sure the ranking makes sense



\*If a user does not exist, but was in an activity, it will not load due to #80 


EDIT:

This now includes the fix for #75 

To test:
1. Create an activity or open one you own
2. Delete it
3. The window should close (sometimes it doesn't, for some reason)
4. There should be no blank cells except possibly at the bottom.
